azkaban-web-server/src/main/java/cloudflow/daos/ProjectDaoImpl.java

```diff
@@ -0,0 +1,175 @@
+      return transOperator.getLastInsertId();
+    };
+
+    String projectId = "";
+    try {
+      projectId = dbOperator.transaction(insertAndGetProjectId).toString();
+      if (projectId.isEmpty()) {
+        throw new ProjectManagerException("Insert project for '" + name + "' failed.");
+      }
+      projectAdminDaoImpl.addAdmins(projectId, project.getAdmins());
+    } catch (SQLException ex) {
+      log.error(INSERT_PROJECT + " failed with exception:", ex);
+      throw new ProjectManagerException("Insert project for '" + name + "' failed.",
+          ex);
+    }
+    return projectId;
+  }
+
+  @Override
+  public Optional<Project> get(String projectId) {
+    List<Project> projects = new ArrayList<>();
+    FetchProjectHandler fetchProjectHandler = new FetchProjectHandler();
+    try {
+      projects = dbOperator.query(FetchProjectHandler.SELECT_PROJECT_WITH_ID, fetchProjectHandler,
+          projectId);
+      for(Project project : projects) {
+        project.setAdmins(projectAdminDaoImpl.findAdminsByProjectId(project.getId()));
+      }
+    } catch (SQLException ex) {
+      log.error("Select for project record with id " + projectId + " failed: ", ex);
+    }
+    return projects.isEmpty() ? Optional.empty() : Optional.of(projects.get(0));
+  }
+
+  @Override
+  public List<Project> getAll(User user) {
+    List<Project> projects = new ArrayList<>();
+    FetchProjectHandler fetchProjectHandler = new FetchProjectHandler();
+
+    /* Get all projects created or modified by user*/
+    try {
+      projects = dbOperator.query(FetchProjectHandler.SELECT_ALL_PROJECTS_BY_USER,
+          fetchProjectHandler, user.getUserId(), user.getUserId());
+      for(Project project : projects) {
+        project.setAdmins(projectAdminDaoImpl.findAdminsByProjectId(project.getId()));
+      }
+    } catch (SQLException ex) {
+      log.error("Get all projects for user '" + user.getUserId() + "'failed: ", ex);
+    }
+
+    return projects;
+  }
+
+
+  public static class FetchProjectHandler implements ResultSetHandler<List<Project>> {
+
+    static String SELECT_PROJECT_WITH_ID =
+        "SELECT id, name, active, modified_time, create_time, version, last_modified_by, "
+            + "description, enc_type, settings_blob, space_id, created_by FROM projects WHERE id"
+            + " = ? and active = 1";
+
+    static String SELECT_PROJECT_WITH_NAME_AND_SPACE_ID =
+        "SELECT id, name, active, modified_time, create_time, version, last_modified_by, "
+            + "description, enc_type, settings_blob, space_id, created_by FROM projects"
+            + " WHERE name = ? and space_id = ? and active = 1";
+
+    static String SELECT_ALL_PROJECTS_BY_USER =
+        "SELECT id, name, active, modified_time, create_time, version, last_modified_by, "
+            + "description, enc_type, settings_blob, space_id, created_by FROM projects WHERE "
+            + "created_by = ? OR last_modified_by = ? and active = 1";
```

Comment:
Some of the fields requested are not actually used. Do we still want to keep them?

Reply:
I would prefer to keep them for now.